server/api/blueprints/mcdm.py

```diff
@@ -1,11 +1,117 @@
-from flask import Blueprint
+from pydoc import doc
+from flask import Blueprint, request, session
+from flask_api import status
 
-mcdm = Blueprint("mcdm", __name__, url_prefix="/mcdm")
+from api.database import (
+    get_scorecards,
+    add_winners
+)
+
+class algorithm:
+
+    def __init__(self,inp):
+        self.ourCalculations = {}
+        self._grades = [] 
+        self.category = inp
+
+    def grades(self):
+        return self._grades
+
+    def get_calculation(self,inputt,grade):
+        if (inputt,grade) in self.ourCalculations.keys():
+            a = self.ourCalculations[(inputt,grade)]
+        else:
+            a = 0
+        return(a)
+                
+    def set_calculation(self,a,inputt,grade):
+        self.ourCalculations[(inputt,grade)] = a
+        if not grade in self._grades:
+            self._grades.append(grade)
+
+    def set_weight(self,weights, studentID):
+        for inputt in self.category:
+            calculated_score = sum([weights[grade]*self.get_calculation(inputt,grade) for grade in weights.keys()])
+            self.set_calculation(calculated_score, inputt,studentID)
+            
+    def setDictionary(self,calculation_array):
+        for (inputt,grade) in calculation_array.keys():
+            self.set_calculation(calculation_array[(inputt,grade)], inputt,grade)
 
+mcdm = Blueprint("mcdm", __name__, url_prefix="/mcdm")
 
-@mcdm.route("/", methods=["GET"])
+@mcdm.route("/test", methods=["POST"])
 def run_mcdm():
     """
     get /mcdm/schoolarship/<id>/: runs the mcdm algorithm, including querying for all scorecards
+
+    sample input:
+
+    {
+    "scholarship_id": "6248d7666a62d69298a0083c",
+    "number_of_awards": "5"
+    }
+
+
     """
-    return
+    
+    form = request.get_json()
+    scholarship_id = form.get("scholarship_id")
```

Comment:
You can run a query for the scholarship with this given id. Then you'll be able to get the number of awards (without me supplying it in the POST request body) and the weightings.

Reply:
Done